core/src/main/java/org/wso2/msf4j/interceptor/MSF4JRequestInterceptor.java

```diff
@@ -0,0 +1,71 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.msf4j.interceptor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.msf4j.Request;
+import org.wso2.msf4j.Response;
+
+/**
+ * Interface that needs to be implemented to intercept request method calls
+ * <p>
+ * Priority of request interceptors will be
+ * Global request interceptors -> Class level annotated request interceptors
+ * -> Method level annotated interceptors -> Resource method execution
+ * -> sub-resource method class level annotated interceptors -> sub-resource method method level annotated interceptors
+ * -> sub-resource execution
+ * <p>
+ * Upon returning false from any interceptor will cause the above flow to break and send the response immediately
+ * <p>
+ * Upon exception in interceptors "onRequestInterceptionError" default method in the interface will be called.
+ * Override this method for custom behaviour
+ */
+@FunctionalInterface
+public interface MSF4JRequestInterceptor {
```

Comment:
If we change the interface name to a specific name, we need to change the annotation name also to a similar name so that the usage will be clear for the end developer. The only way we avoid having the same name for the 2 classes is by adding the "MSF4J" prefix. Furthermore having the same name for the annotation and the interface will result in:

- Confusing the end developer (readability reasons)
- When you are using both, the annotation and the class, one of them will have a longer name